src/lib/header-bidding/prebid-types.ts

```diff
@@ -172,32 +169,176 @@ export type PrebidBidderErrorEvent = {
 	};
 };
 
-export type PrebidAuctionInitEvent = {
+type PrebidAuctionInitEvent = {
 	eventType: 'auctionInit';
 	args: {
 		adUnitCodes: string[];
 		bidderRequests: PrebidBidderRequest[];
 	};
 };
 
-export type PrebidAuctionEndEvent = {
+type PrebidAuctionEndEvent = {
 	eventType: 'auctionEnd';
 	args: {
 		adUnitCodes: string[];
 		bidsReceived: PrebidBid[];
 	};
 };
 
-export type PrebidEvent =
+type PrebidEvent =
 	| PrebidBidReqestedEvent
 	| PrebidBidderErrorEvent
 	| PrebidAuctionInitEvent
 	| PrebidAuctionEndEvent;
 
-export type PrebidMediaTypes = {
+type PrebidMediaTypes = {
 	banner: {
 		sizes: HeaderBiddingSize[];
 	};
 };
 
-export type SlotFlatMap = (slot: HeaderBiddingSlot) => HeaderBiddingSlot[];
+type SlotFlatMap = (slot: HeaderBiddingSlot) => HeaderBiddingSlot[];
+
+interface AnalyticsAdapterConfig {
+	analyticsType: string;
+}
+
+interface BidArgs {
+	bidderCode?: string;
+	adserverTargeting?: Record<string, string>;
+	adId?: string;
+	requestId?: string;
+	auctionId?: string;
+	bidder?: string;
+	bidId?: string;
+	bids?: Array<{
+		adUnitCode?: string;
+		bidId?: string;
+	}>;
+	adUnitCode?: string;
+	cpm?: number;
+	pbCg?: number;
+	currency?: string;
+	netRevenue?: boolean;
+	creativeId?: string;
+	size?: string;
+	timeToRespond?: number;
+	dealId?: string;
+	statusMessage?: string;
+	start?: number;
+	meta?: {
+		networkId?: string;
+		buyerId?: string;
+		brandId?: string;
+		brandName?: string;
+		clickUrl?: string;
+	};
+}
+
+type TrackParams = {
+	eventType: string;
+	args: BidArgs;
+};
+
+const eventKeys = [
```

Comment:
Would be nice to potentially have comments explaining what these stand for, as I can imagine that coming up in future, but could be a follow up!

Reply:
Good shout, there's a comment in the data platform models that I could steal them from actually!